### ISSUE TITLE
Support Python dispatching on PrimaryInputName

### ISSUE THREAD

**Issue by thewtex** [InsightSoftwareConsortium/ITK#4858]
We should dispatch on PrimaryInputName when it is passed as a keyword argument instead of a positional parameter.

xref:

- https://github.com/SuperElastix/elastix/blob/368d71649c9052b47396b9eb7c0069fa7ea24b7f/Core/Main/itkElastixRegistrationMethod.hxx#L55
- https://github.com/InsightSoftwareConsortium/ITKElastix/issues/212

CC: @N-Dekker @mstaring

**Comment by N-Dekker** [comment]
@thewtex I'm glad you are looking at this issue. Is the following supposed to work well?

```python
    output_image = itk.median_image_filter(input=my_float_image)
```
(It now says _TypeError: Expecting argument of type itkImageSS2 or itkImageSourceISS2_). Or should the keyword argument start with a capital, `Input=my_float_image`? I'm asking, because "Input" is in the list of so-called `primary_input_methods` at:

https://github.com/InsightSoftwareConsortium/ITK/blob/15af3aed65693811448c9af22ce9d09ff9f3000a/Wrapping/Generators/Python/itk/support/template_class.py#L672

And `primary_input_methods`  is used for dispatching:

https://github.com/InsightSoftwareConsortium/ITK/blob/15af3aed65693811448c9af22ce9d09ff9f3000a/Wrapping/Generators/Python/itk/support/template_class.py#L711-L716

Regarding ITKElastix issue https://github.com/InsightSoftwareConsortium/ITKElastix/issues/212, obviously, "FixedImage" and "MovingImage" are not in `primary_input_methods` (and neither are "fixed_image" and "moving_image"). So I guess `primary_input_methods` should be extended, right?

**Comment by blowekamp** [comment]
The Process object maintains a lists of index inputs and named inputs. And a named input can be marked as primary. These named inputs are already coded input many filters e.g.
[PDEDeformableRegistrationFilter](https://github.com/InsightSoftwareConsortium/ITK/blob/15af3aed65693811448c9af22ce9d09ff9f3000a/Modules/Registration/PDEDeformable/include/itkPDEDeformableRegistrationFilter.hxx#L37-L46)

The information in the Process object should likely be used.

**Comment by N-Dekker** [comment]
Thanks @blowekamp I see, PDEDeformableRegistrationFilter has inputs named "FixedImage" and "MovingImage":

https://github.com/InsightSoftwareConsortium/ITK/blob/15af3aed65693811448c9af22ce9d09ff9f3000a/Modules/Registration/PDEDeformable/include/itkPDEDeformableRegistrationFilter.hxx#L42-L46 

So then, should something like the following work, for `itk::Image<float>` objects `float_image1` and `float_image2`? 

```python
itk.pde_deformable_registration_filter(FixedImage=float_image1, MovingImage=float_image2)
```

(It now says _TypeError: Expecting argument of type itkImageSS2 or itkImageSourceISS2_)

**Comment by dzenanz** [comment]
In procedural interface, the parameters get converted to `snake_case`, so try this: `itk.pde_deformable_registration_filter(fixed_image=float_image1, moving_image=float_image2)`.

**Comment by blowekamp** [comment]
I would expect [ProcessObject::GetPrimaryInputName](https://itk.org/Doxygen413/html/classitk_1_1ProcessObject.html#a8f97b6176f97ad8d8b3fc6d1edb70520) to be useful.  

I am not too sure of these details in ITK Python. Specifically if the Process object's name inputs/outputs are used or if it's just mapping keyword arguments to methods.

Also I think that PDE deformable registration class may be a base class for the deacons methods, and I have a harder guess of which is the primary input, and how the template parameters get mapped from it.

**Comment by thewtex** [comment]
> Is the following supposed to work well?
> 
> ```python
>     output_image = itk.median_image_filter(input=my_float_image)
> ```

The keyword here to be supported would be the snake_case version of `itk::MedianImageFilter::GetPrimaryInputName()`;

**Comment by thewtex** [comment]
Closed via #4921
